tests/Types/RealFloatTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Tests\Types;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\RealFloatType;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class RealFloatTest extends TestCase
+{
+    private AbstractPlatform&MockObject $platform;
+    private RealFloatType $type;
+
+    protected function setUp(): void
+    {
+        $this->platform = $this->createMock(AbstractPlatform::class);
+        $this->type     = new RealFloatType();
+    }
+
+    public function testFloatConvertsToPHPValue(): void
+    {
+        self::assertIsFloat($this->type->convertToPHPValue('5.5', $this->platform));
```

Comment:
Are we sure we want to accept _any_ float here?

Reply:
I think yes, for PHP there's no difference in float types. Although this test may be redundant because a string shouldn't come from the database in any case. If I understand you correctly.

Reply:
> I think yes, for PHP there's no difference in float types.

So, if `convertToPHPValue('5.5', $platform)` returns `42.3`, that would be totally fine for us because after all, it's a float? 🤨  I would've expected something very close to `5.5`. 🙂 

> Although this test may be redundant because a string shouldn't come from the database in any case.

Not sure if there's a driver that still returns floats as strings nowadays, but I think it's still good that we handle the case.